research_daps/orms/glass.py

```diff
@@ -0,0 +1,202 @@
+#     # TODO: Rank Integer column indicating address rank
+#     # TODO: Date Date column indicating date of latest dump with this relationship
+# )
+class OrganisationAddress(Base):
+    """Association object between organisations and addresses"""
+
+    org_id = Column(Integer, ForeignKey("organisation.org_id"), primary_key=True)
+    address_id = Column(Integer, ForeignKey("address.address_id"), primary_key=True)
+    # TODO: do we really want to keep ranks? If ranks constantly change then
+    # this proliferates a lot of data
+    rank = Column(
+        Integer, nullable=False, index=True, doc="Rank of `address_id` for `org_id`"
+    )
+    is_active = Column(
+        Boolean,
+        nullable=False,
+        index=True,
+        doc="Indicates whether `address_id` is active",
+    )
+    # TODO : should this really have two date columns: valid_from and valid_to ?
+    date = Column(
+        Date,
+        nullable=False,
+        index=True,
+        doc="Date of data-dump associating org with address",
+    )
+    organisation = relationship("Organisation", back_populates="addresses")
+    address = relationship("Address", back_populates="organisations")
+
+
+organisation_sector = Table(
+    "organisation_sector",
+    Base.metadata,
+    Column("org_id", Integer, ForeignKey("organisation.org_id")),
+    Column("sector_id", Integer, ForeignKey("sector.sector_id")),
+    # TODO: Active? boolean column indicating whether sector is in latest data dump
+    # TODO: Rank Integer column indicating sector rank
+    # TODO: Date Date column indicating date of latest dump with this relationship
+)
+
+notice_terms = Table(
+    "notice_terms",
+    Base.metadata,
+    Column("notice_id", Integer, ForeignKey("notice.notice_id")),
+    Column("term_id", Integer, ForeignKey("covid_term.term_id")),
+)
+
+
+class Organisation(Base):
+    """An organisation relating to a business website"""
+
+    org_id = Column(Integer, primary_key=True)
+    name = Column(
+        String,
+        unique=True,
+        doc="Organisation name inferred by named entity recognition",
+        index=True,
+    )
+    website = Column(String, nullable=False, doc="URL of organisation")
+    active = Column(
+        Boolean,
+        nullable=False,
+        doc="True if Organisation was contained in last data-dump",
+    )
+    # One-to-many
+    notices = relationship("Notice", backref=backref("organisation"))
+    descriptions = relationship(
+        "OrganisationDescription", backref=backref("organisation")
+    )
+    metadata_ = relationship("OrganisationMetadata", backref=backref("organisation"))
+    # Many-to-many
+    sectors = relationship(
+        "Sector", secondary=organisation_sector, back_populates="organisations"
+    )
+    addresses = relationship("OrganisationAddress", back_populates="organisation")
+    # addresses = relationship(
+    #     "Address", secondary=organisation_address, back_populates="organisations"
+    # )
```

Comment:
Do you also want to add in a `relationship` to `OrganisationCompaniesHouseMatch` here?